src/lib/Tests/Form/DataTransformer/ContentInfoTransformerTest.php

```diff
@@ -0,0 +1,145 @@
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Tests\Form\DataTransformer;
+
+use eZ\Publish\API\Repository\ContentService;
+use PHPUnit\Framework\TestCase;
+use eZ\Publish\API\Repository\Values\Content\ContentInfo;
+use EzSystems\EzPlatformAdminUi\Form\DataTransformer\ContentInfoTransformer;
+use Symfony\Component\Form\Exception\TransformationFailedException;
+use eZ\Publish\API\Repository\Exceptions\NotFoundException;
+use Throwable;
+
+class ContentInfoTransformerTest extends TestCase
+{
+    /**
+     * @var ContentInfoTransformer
+     */
+    private $contentInfoTransformer;
+
+    public function setUp()
+    {
+        $contentService = $this->createMock(ContentService::class);
+        $contentService->expects(self::any())
+            ->method('loadContentInfo')
+            ->with(123456)
+            ->willReturn(new ContentInfo(['id' => 123456]));
+
+        $this->contentInfoTransformer = new ContentInfoTransformer($contentService);
+    }
+
+    /**
+     * @dataProvider transformDataProvider
+     * @param $value
+     * @param $expected
+     */
+    public function testTransform($value, $expected)
+    {
+        $result = $this->contentInfoTransformer->transform($value);
+
+        $this->assertEquals($expected, $result);
+    }
+
+    /**
+     * @dataProvider transformWithInvalidInputDataProvider
+     * @param $value
+     */
+    public function testTransformWithInvalidInput($value)
+    {
+        $this->expectException(TransformationFailedException::class);
+        $this->expectExceptionMessage('Expected a ' . ContentInfo::class . ' object.');
+        $this->contentInfoTransformer->transform($value);
+    }
+
+    /**
+     * @dataProvider reverseTransformDataProvider
+     * @param $value
+     * @param $expected
+     */
+    public function testReverseTransform($value, $expected)
+    {
+        $result = $this->contentInfoTransformer->reverseTransform($value);
+
+        $this->assertEquals($expected, $result);
+    }
+
+    /**
+     * @dataProvider reverseTransformWithInvalidInputDataProvider
+     * @param $value
+     */
+    public function testReverseTransformWithInvalidInput($value)
+    {
+        $this->expectException(TransformationFailedException::class);
+        $this->expectExceptionMessage('Expected an integer.');
+        $this->contentInfoTransformer->reverseTransform($value);
+    }
+
+    public function testReverseTransformWithNotFoundException()
+    {
+        $this->expectException(TransformationFailedException::class);
+        $this->expectExceptionMessage('Transformation failed. Language not found');
+
+        $service = $this->createMock(ContentService::class);
+        $service->method('loadContentInfo')
+            ->will($this->throwException(new class() extends NotFoundException {
```

Comment:
Why are you creating an exception class here? Can't you use the one coming form kernel package, can you? If you have to I'd create separate class inside `Tests/` folder. This looks terrible :P